codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSEndpointTraitTransformer.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.aws.swift.codegen.model
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.traits.EndpointTrait
+import software.amazon.smithy.model.transform.ModelTransformer
+import software.amazon.smithy.rulesengine.traits.StaticContextParamsTrait
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.getTrait
+
+/**
+ * This integration is responsible for removing the EndpointTrait from the operation
+ * - For S3 Control, if hostPrefix is {AccountId}., then remove the Endpoint Trait because it is already handled
+ *      within the EndpointResolver
+ */
+class AWSEndpointTraitTransformer : SwiftIntegration {
```

Comment:
This and the other trait transformer have a lot of pretty deeply nested logic... should we break down the logic a bit more?

Reply:
I wouldn't because it is very specific to case, breaking it would make understanding it harder than at just one place.